www/docs/migration/migrate-from-v9-to-v10.mdx

```diff
@@ -1,257 +1,67 @@
+Nearly all of the concepts from tRPC v9 will carry over to tRPC v10. Most of the changes for v10 are syntax based.
 
-In a gist, what has changed?
+Under the hood, we are unlocking performance improvements, bringing you quality of life enhancements, and creating room for us to build new features in the future.
 
-### The `t` variable
+tRPC v10 features a compatibility layer for users coming from v9. `.interop()` allows you to incrementally adopt v10 so that you can continue building the rest of your project while still enjoying v10's new features.
 
-> If you don't like the variable name `t`, you can call it whatever you want
+:::caution
+There are [a few features that are not supported by `.interop()`](#limitations-of-interop). We expect nearly all of our users to be able to use `.interop()` to migrate their server side code in only a few minutes. If you are discovering that `.interop()` is not working correctly for you, be sure to [check here](#limitations-of-interop).
+:::caution
 
-The way you initialize tRPC on the server has been update, we now create a root `t` variable that contains the root information about your app:
+## Enabling `interop()` on a v9 router
 
-- [Context](/docs/context)
-- [Meta data](/docs/metadata)
-- [Error formatter](/docs/error-formatting)
-- [Data Transformer](/docs/error-formatting)
+Turning your v9 router into a v10 router only takes 10 characters. Add `.interop()` to the end of your v9 router...and you're done with your server code!
 
-The way the `t` variable is defined is simply like this:
+```ts title="/server.ts"
+import * as trpc from '@trpc/server';
 
-```ts title='/src/server/trpc.ts'
-import { initTRPC } from '@trpc/server';
-
-// Beware of the double `()()`
-export const t = initTRPC()();
-```
-
-Here's a full example of how the `t` variable may look like with a data-transformer, openapi metadata and error formatter:
-
-```ts title='/src/server/trpc.ts'
-import { initTRPC } from '@trpc/server';
-import superjson from 'superjson';
-
-// This is usually inferred
-interface Context {
-  user?: {
-    id: string;
-    name: string;
-  };
-}
-
-interface Meta {
-  openapi: {
-    enabled: boolean;
-    method: string;
-    path: string;
-  };
-}
-
-export const t = initTRPC<{
-  ctx: Context;
-  meta: Meta;
-}>()({
-  errorFormatter({ shape, error }) {
-    return {
-      ...shape,
-      data: {
-        ...shape.data,
-        zodError:
-          error.code === 'BAD_REQUEST' &&
-          error.cause instanceof ZodError
-            ? error.cause.flatten()
-            : null,
-      };
-    };
-  },
-  transformer: superjson,
-})
-```
-
-### Defining routers & procedures
-
-```ts
-// OLD:
-const appRouter = trpc.router()
-  .query('greeting', {
-    input: z.string(),
-    resolve({ input }) {
-      return `hello ${input}!`;
-    },
-  });
-
-// NEW:
-const appRouter = t.router({
-  greeting: t.procedure
-    .input(z.string())
-    .query(({ input }) => `hello ${input}!`),
-});
-```
```

Comment:
I really liked having the old and new side-to-side, how come that was deleted?

Reply:
I guess this belongs more in a blog article - was that your thinking? 🤔

Reply:
This does indeed belong. I think I cut it as a part of the larger block that was demonstrating too much - but the stanzas you've marked here are certainly worthy.

I'll find a place for these. My first instinct is first section of the page - but will see once I pop it back open on my machine.